### DISCUSSION TITLE
Are 2 and more battery packs supported?

### DISCUSSION THREAD

**Discussion by linuxdep** [syssi/esphome-seplos-bms#40] [General]
Hello, since my batteries are still waiting to be moved into the housing, I ended up looking here.
Is it possible to read and control all data from the BMS with the project? Above all, are several such packs that are connected together via CAN on the WR hanging also supported? Or is only a single BMS possible?

**Comment by syssi** [comment]
I hope one of the active users can help here. In general this component supports to request the data from different BMSs (per address of the dip switch) on the bus. If I remember/understand correctly a Seplos setup changes the behavior if you declare one device as master and others as slave. In this case the data (f.e. the state of charge) of the different batteries is aggregated and presented as a single battery (state of charge).

In other words: If you attach an inverter via CAN to a chained Seplos setup I don't know it's still possible to query the individual values per BMS or the aggregate only via RS485.

**Reply by syssi** [reply]
This is the configuration example to communicate with 3 Seplos devices on the same bus:

https://github.com/syssi/esphome-seplos-bms/blob/main/esp8266-example-multiple-battery-banks.yaml

**Reply by syssi** [reply]
Some more infos about the topic: https://github.com/syssi/esphome-seplos-bms/issues/30

**Comment by linuxdep** [comment]
Thanks. I had already read some of them there, thought there would be new insights from users with +2 batteries and a new BMS version

Wouldn't it also be conceivable not to communicate via RS485 but via CAN bus? Then the problem with the invisible master would no longer exist, right?

**Reply by syssi** [reply]
If I understand the Seplos CAN bus protocol details correctly just the aggregated data is available here. Furthermore the CAN bus support of ESPHome isn't stable at the moment.

**Reply by linuxdep** [reply]
ok, Then I want to hope that seplos has solved the problem with the new hardware and all data is visible, otherwise only BT remains.
do you have a document for seplos CAN communication? Or BT protocol?

**Reply by syssi** [reply]
Just google "Seplos CAN bus protocol". This BMS is available with bluetooth support?

**Comment by linuxdep** [comment]
yes, 10E is with BT. 
Protocol 1.0 found, there is really little information transmitted, but it is enough for a WR.

**Reply by syssi** [reply]
Most Inverters are interested in SoC (state of charge), SoH (state of health) and the total battery voltage only.

**Comment by Meli73** [comment]
In near future I also want to double my 10E 16S BMS with a 2nd Pack/BMS.  

Seplos told me yesterday also,  I only could try BT to get the Master BMS Values :-(

**Reply by Meli73** [reply]
But even it could be possible - I'm not so happy with this wireless option by BT.

Strange new Idea:  The Problem is the Master BMS.  What would be, if there is no Master at all ? Only Slave BMS -  and the ESP home controller doing the Master Job talking to the CAN counterpart

* It needs beside RS485 also CAN hardware as 2nd interface to the Uplink inverter System
* It needs to know the "CAN Dialekt" / protocol of the present inverter (Victron Multiplus II at my instalation)  

Should be possible? or ?????

**Reply by syssi** [reply]
You are on the right track. The idea should be possible in general. I've started this project in the past

https://github.com/syssi/esphome-virtual-can-bms

to fill the mentioned gap (passing computed measurements via CAN bus to an inverter). This project is incomplete and untested because there aren't any users / testers yet. I don't own a inverter with CAN bus support.

**Reply by Meli73** [reply]
Wow .. great :-)   how could I help ?  

What Can hardware do you support ?

Would it help to write some Log's of CAN traffic?  and RS485 read out at the same time ?

at frist at a normal "one-BMS" system.    and in the 2nd step at a running Master/Slave konfiguration ?

**Reply by syssi** [reply]
A traffic capture of the CAN traffic would be nice to get an idea which measurements gets exchanged at your setup.

The mentioned project implements some registers of the SMA protocol:

https://github.com/syssi/esphome-virtual-can-bms/blob/main/docs/SMA%20CAN%20protocol.pdf
https://github.com/syssi/esphome-virtual-can-bms/blob/main/components/virtual_can_bms/virtual_can_bms.cpp#L20-L23

In best case this will match with your setup. In worst case we have to add some new messages.

https://www.victronenergy.com/upload/documents/Technical-Information-Data-communication-with-Victron-Energy-products_EN.pdf

**Reply by syssi** [reply]
Let's continue here: https://github.com/syssi/esphome-virtual-can-bms/issues/3

**Comment by vejvik** [comment]
I have two MASON-280 battery boxes with BMS 1101-10E-SP76-16S. From the master, will be connected CAN to the inverter, and from the master, rs4851 will be connected to slave rs4852. (DIP switches according to CAN 2p connection)
How could I read the data via esp to HomeAssistant, how would the wiring diagram look like?
thank you for the advice
